src/e2e-test/java/io/cdap/plugin/BQValidation.java

```diff
@@ -0,0 +1,229 @@
+import java.util.List;
+import java.sql.*;
+
+/**
+ * BQValidation.
+ */
+public class BQValidation {
+
+  static List<JsonObject> bigQueryResponse = new ArrayList<>();
+  static List<Object> bigQueryRows = new ArrayList<>();
+  static Gson gson = new Gson();
+
+  public static void main(String[] args) throws IOException, InterruptedException, SQLException, ParseException {
+//    List<Object> l=new ArrayList<>();
+//
+//    getBigQueryTableData("E2E_SOURCE_7ff9f7fc_bf6c_46d1_a050_1d4525e7b575",l);
+//    for(int i=0;i<l.size();i++)
+//    {
+//      System.out.println(l.get(i));
+//    }
+    //validateSalesforceToBQRecordValues(SObjects.AUTOMATION_CUSTOM__C.value, "myTable7");
+    validateBQToSalesforceRecordValues("E2E_SOURCE_7ff9f7fc_bf6c_46d1_a050_1d4525e7b575",SObjects.AUTOMATION_CUSTOM__C.value);
+  }
+
+
+  public static boolean validateSalesforceToBQRecordValues(String objectName, String targetTable)
+    throws SQLException, IOException, InterruptedException, ParseException {
+    String Id = "a03Dn000008bkucIAA";
+    //big query table="myTable7";
+
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject json = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(json);
+    }
+    SalesforceClient.queryObject(Id, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.objectResponseList, bigQueryResponse);
+
+  }
+  public static boolean validateBQToSalesforceRecordValues(String targetTable,String objectName)
+    throws SQLException, IOException, InterruptedException, ParseException {
+    String Id = "a03Dn000008c4D8IAI";
+    //big query table="E2E_SOURCE_7ff9f7fc_bf6c_46d1_a050_1d4525e7b575";
+
+    getBigQueryTableData(targetTable, bigQueryRows);
+    for (Object rows : bigQueryRows) {
+      JsonObject json = gson.fromJson(String.valueOf(rows), JsonObject.class);
+      bigQueryResponse.add(json);
+    }
+    SalesforceClient.queryObject(Id, objectName);
+    return compareSalesforceAndJsonData(SalesforceClient.objectResponseList, bigQueryResponse);
+
+  }
+
+  private static void getBigQueryTableData(String table, List<Object> bigQueryRows)
+    throws IOException, InterruptedException {
+
+    String projectId = PluginPropertyUtils.pluginProp("projectId");
+    String dataset = PluginPropertyUtils.pluginProp("dataset");
+    String selectQuery = "SELECT TO_JSON(t) FROM `" + projectId + "." + dataset + "." + table + "` AS t";
+    TableResult result = BigQueryClient.getQueryResult(selectQuery);
+    result.iterateAll().forEach(value -> bigQueryRows.add(value.get(0).getValue()));
+  }
+
+  public static boolean compareSalesforceAndJsonData(List<JsonObject> salesforceData,
+                                                     List<JsonObject> bigQueryData)
+    throws SQLException, ParseException {
+    boolean result = false;
+    if (bigQueryData == null) {
+      Assert.fail("bigQueryData is null");
+      return result;
```

Comment:
if bigQueryData is null. Assert.fail is sufficient. Please remove return result statement.